test/replication-luatest/election_fencing_test.lua

```diff
@@ -289,3 +282,80 @@ g_sync.test_fencing = function(g)
     luatest.assert_equals(ret, {},
         'Sync write rollbacked after new leader discovered')
 end
+
+g_mode.before_all(function(g)
+    start(g)
+    box_cfg_update({g.server_3}, {replication = {}})
+    wait_disconnected(g.server_1, g.server_3)
+    wait_disconnected(g.server_2, g.server_3)
+end)
+
+g_mode.after_all(stop)
+
+g_mode.test_fencing_mode = function(g)
+    local timeout = 0.5
+    box_cfg_update({g.server_1, g.server_2}, {
+        election_fencing_mode = g.params.election_fencing_mode,
+        replication_timeout = timeout,
+    })
+
+    g.server_1:exec(function(input, output)
+        rawset(_G, 'proxy', require('test.luatest_helpers.proxy.proxy'):new({
+            input_path = input,
+            output_path = output,
+        }))
+        _G.proxy:start({force = true})
+    end, {
+        server.build_instance_uri('server_1_proxy'),
+        server.build_instance_uri('server_1')})
+
+    local proxied_replication = {
+        server.build_instance_uri('server_1_proxy'),
+        server.build_instance_uri('server_2'),
+    }
+
+    box_cfg_update({g.server_2}, {replication = {}})
+    wait_disconnected(g.server_1, g.server_2)
+    box_cfg_update({g.server_2}, {replication = proxied_replication})
+    wait_connected(g.server_1, g.server_2)
+
+    promote(g.server_1)
+    local leader_id = g.server_1:instance_id()
+    wait_sync(g.server_1, {g.server_2})
+
+    box_cfg_update({g.server_2}, {
+        election_mode = 'candidate',
+        replication_synchro_quorum = 1,
+    })
+
+    g.server_1:exec(function()
+        _G.proxy:pause()
+        for _, c in ipairs(_G.proxy.connections) do
+            c:pause()
+        end
```

Comment:
Seems this piece of code will be used a lot. Maybe add a helper for it?
I would suggest merging `proxy:pause()` and `proxy:resume()` to `proxy:accept_new_connections( true/false)`
and make
`proxy:pause()` do something like
```lua
proxy:accept_new_connections(false)
for _, c in pairs(proxy.connections) do
    c:pause()
end
```

Reply:
Thanks for review! I've reworked start/pause/resume/stop methods, take a look.

Reply:
Thanks! That's what I wanted.